### PR TITLE
Workflow with a preview collection (fixes #126)

### LINKED ISSUE #126: Preview signed collection when status = to-review
As suggested by Travis and @chartjes , the use-case is the following:

* Someone edits the staging collection and asks for a review
* A preview collection now contains the new changes and the signature
* QA sets their scripts/clients towards the preview collection (by changing the Firefox internal pref for example)
* QA approves changes and sets collection to `to-sign`
* The production collection now contains the new changes and every client will obtain them

### PR DESCRIPTION
**Pull request by leplatrem**
Fixes #126 

* [x] Draft of documentation/schema
* [x] Switch resources parsing to triplet ( `(source, None, destination)` if two provided)
* [x] Generalize the signing + copy of `to-sign` to `to-review`
* [x] Implement functional test
* [x] Update changelog

@tarekziade r?

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -185,6 +185,20 @@ makes sure that:
 
 See `Kinto groups API <http://kinto.readthedocs.io/en/stable/api/1.x/groups.html>`_ for more details about how to define groups.
 
+If the review process is enabled, it is possible to configure a *preview*
+collection, that will be updated and signed when the status is set to ``to-review``.
+This *preview* collection can be used by client to test and validate the changes
```
Comment (tarekziade) [inline README.rst]: s/client/clients

File: README.rst [hunk 2]
```diff
@@ -185,6 +185,20 @@ makes sure that:
...
+before approving them.
+
+If a resources entry contains a semi-column separated **triplet**, then a preview
+collection will be enabled.
+
+.. code-block:: ini
+
+  kinto.signer.resources =
+      /buckets/staging/collections/articles;/buckets/preview/collections/articles;/buckets/blog/collections/articles
```
Comment (tarekziade) [inline README.rst]: just an proposal,  feel free to ignore, but could we use  "=>"  instead of ";" to show the worklfow

File: README.rst [hunk 3]
```diff
@@ -345,6 +359,12 @@ To generate a new keypair, you can use the following command::
 Running the tests
 =================
 
+In order to contribute and run the full functional test suite locally you need
```
Comment (tarekziade) [inline README.rst]: why ? can't we use the latest binary of autograph ?

File: kinto_signer/updater.py [hunk 4]
```diff
@@ -310,6 +327,8 @@ def update_source_status(self, status, request):
         attrs = {'status': status.value}
         if status == STATUS.WORK_IN_PROGRESS:
             attrs["last_author"] = request.prefixed_userid
+        if status == STATUS.TO_REVIEW:
+            attrs["last_editor"] = request.prefixed_userid
```
Comment (tarekziade) [inline kinto_signer/updater.py]: do we want last_author, last_editor, last_reviewer to be constants ?

File: README.rst (hunk 3 above)
Comment (leplatrem) [inline README.rst]: we use `go get`

File: README.rst (hunk 2 above)
Comment (leplatrem) [inline README.rst]: Good idea! I added it to #88

File: README.rst (hunk 3 above)
Comment (tarekziade) [inline README.rst]: In that case we should pin the autograph version we use to the latest release I think.

**Comment by tarekziade** [comment]
r+

**Pull request merged**